substrate/frame/system/src/lib.rs

```diff
@@ -955,23 +955,16 @@ pub mod pallet {
 
 	/// Events deposited for the current block.
 	///
-	/// NOTE: The item is unbound and should therefore never be read on chain.
+	/// NOTE: The map is unbound and should therefore never be iterated on chain.
 	/// It could otherwise inflate the PoV size of a block.
 	///
 	/// Events have a large in-memory size. Box the events to not go out-of-memory
 	/// just in case someone still reads them from within the runtime.
 	#[pallet::storage]
-	#[pallet::whitelist_storage]
 	#[pallet::disable_try_decode_storage]
 	#[pallet::unbounded]
 	pub(super) type Events<T: Config> =
-		StorageValue<_, Vec<Box<EventRecord<T::RuntimeEvent, T::Hash>>>, ValueQuery>;
-
-	/// The number of events in the `Events<T>` list.
-	#[pallet::storage]
-	#[pallet::whitelist_storage]
-	#[pallet::getter(fn event_count)]
-	pub(super) type EventCount<T: Config> = StorageValue<_, EventIndex, ValueQuery>;
+		CountedStorageMap<_, Identity, u32, Box<EventRecord<T::RuntimeEvent, T::Hash>>, OptionQuery>;
```

Comment:
Should we use the type alias here instead?
```suggestion
		CountedStorageMap<_, Identity, EventIndex, Box<EventRecord<T::RuntimeEvent, T::Hash>>, OptionQuery>;
```